tests/unit/mixins/I18n.ts

```diff
@@ -82,25 +83,45 @@ registerSuite('mixins/I18nMixin', {
 				});
 			}
 		},
+		'locale data can be injected by defining an Injector with a registry': {
+			'defaults to the injector data'() {
+				const injector = new Injector({ locale: 'ar', rtl: true });
+				const registry = new Registry();
 
-		'root locale switching': {
-			'Updates when no `locale` property is set'() {
+				registry.defineInjector(INJECTOR_KEY, injector);
 				localized = new Localized();
-				sinon.spy(localized, 'invalidate');
+				localized.__setCoreProperties__({ bind: localized, baseRegistry: registry });
+				localized.__setProperties__({});
 
-				switchLocale('fr');
-
-				assert.isTrue((<any> localized).invalidate.called, 'Widget invalidated.');
+				const result = localized.__render__();
+				assert.strictEqual(result.properties!['lang'], 'ar');
+				assert.strictEqual(result.properties!['dir'], 'rtl');
 			},
 
-			'Does not update when `locale` property is set'() {
+			'does not override property values'() {
+				const injector = new Injector({ locale: 'ar', rtl: true });
+				const registry = new Registry();
+
+				registry.defineInjector(INJECTOR_KEY, injector);
 				localized = new Localized();
-				localized.__setProperties__({ locale: 'en' });
-				sinon.spy(localized, 'invalidate');
+				localized.__setCoreProperties__({ bind: localized, baseRegistry: registry });
+				localized.__setProperties__({ locale: 'fr', rtl: false });
 
-				switchLocale('fr');
+				const result = localized.__render__();
+				assert.strictEqual(result.properties!['lang'], 'fr');
+				assert.strictEqual(result.properties!['dir'], 'ltr');
+			},
+
+			'properties can be registered with helper'() {
+				const registry = new Registry();
+				registerI18nInjector({ locale: 'fr' }, registry);
```

Comment:
we should probably have a test to show that you can change the locale via the injector `set()`?

Reply:
Done!